src/main/java/io/divolte/server/PingHandler.java

```diff
@@ -16,32 +16,49 @@
 
 package io.divolte.server;
 
-import io.undertow.server.HttpServerExchange;
-import io.undertow.util.Headers;
-
 import java.nio.charset.StandardCharsets;
-
 import javax.annotation.ParametersAreNonnullByDefault;
 
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.Headers;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static java.net.HttpURLConnection.HTTP_UNAVAILABLE;
+
 /**
  * Event handler for ping requests.
  *
  * Ping requests are immediately and always responded to with a "pong" text response.
  */
 @ParametersAreNonnullByDefault
-final class PingHandler {
+final class PingHandler implements HttpHandler {
     private static final Logger logger = LoggerFactory.getLogger(Server.class);
+    private boolean shutdown;
```

Comment:
Assuming this is not necessarily set from the thread that reads it, I think you'll want to mark that `volatile`.

Reply:
@friso Nice catch